sonar-plugin-api/src/main/java/org/sonar/api/server/authentication/test/TestIdentityProvider.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.server.authentication.test;
+
+import org.sonar.api.server.authentication.IdentityProvider;
+
+public class TestIdentityProvider implements IdentityProvider {
+
+  private String key;
+  private String name;
+  private String iconPatch;
+  private boolean enabled;
+  private boolean allowsUsersToSignUp;
+
+  @Override
+  public String getKey() {
+    return key;
+  }
+
+  public TestIdentityProvider setKey(String key) {
+    this.key = key;
+    return this;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  public TestIdentityProvider setName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  @Override
+  public String getIconPath() {
+    return iconPatch;
+  }
+
+  public TestIdentityProvider setIconPatch(String iconPatch) {
+    this.iconPatch = iconPatch;
+    return this;
+  }
+
+  @Override
+  public boolean isEnabled() {
+    return enabled;
+  }
+
+  public TestIdentityProvider setEnabled(boolean enabled) {
+    this.enabled = enabled;
+    return this;
+  }
+
+  @Override
+  public boolean allowsUsersToSignUp() {
+    return allowsUsersToSignUp;
+  }
+
+  public TestIdentityProvider setAllowsUsersToSignUp(boolean allowsUsersToSignUp) {
+    this.allowsUsersToSignUp = allowsUsersToSignUp;
+    return this;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00122)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Missing curly brace. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00121)